yt_dlp/utils/traversal.py

```diff
@@ -168,14 +169,43 @@ def apply_key(key, obj, is_last):
                 result = next((v for k, v in obj.groupdict().items() if casefold(k) == key), None)
 
         elif isinstance(key, (int, slice)):
-            if is_iterable_like(obj, collections.abc.Sequence):
+            if is_iterable_like(obj, collections.abc.Sequence) or isinstance(obj, xml.etree.ElementTree.Element):
                 branching = isinstance(key, slice)
                 with contextlib.suppress(IndexError):
                     result = obj[key]
             elif traverse_string:
                 with contextlib.suppress(IndexError):
                     result = str(obj)[key]
 
+        elif isinstance(obj, xml.etree.ElementTree.Element) and isinstance(key, str):
+            xpath, sep, special = key.rpartition('/')
+            has_specials = special.startswith('@') or special == 'text()'
+            if not has_specials:
+                xpath = f'{xpath}{sep}{special}'
+
+            # Allow abbreviations of relative paths, absolute paths error
+            if xpath.startswith('/'):
+                xpath = f'.{xpath}'
+            elif xpath and not xpath.startswith('./'):
+                xpath = f'./{xpath}'
+
+            def apply_specials(element):
+                if special == '@':
+                    return element.attrib
+                if special.startswith('@'):
+                    return try_call(element.attrib.get, args=(special[1:],))
+                if special == 'text()':
+                    return element.text
```

Comment:
```suggestion
                if special == '@':
                    return element.attrib
                elif special.startswith('@'):
                    return try_call(element.attrib.get, args=(special[1:],))
                elif special == 'text()':
                    return element.text
```

just for clarity

Reply:
There is an explicit return and it is only one line per `if`. I don't think you win any clarity by doing `elif`

Reply:
I personally find easier. But ignore if u don't like it